.github/workflows/rust.yml

```diff
@@ -101,3 +102,35 @@ jobs:
       - uses: Swatinem/rust-cache@v1
       - name: Doc
         run: RUSTDOCFLAGS="-Dwarnings" cargo doc --workspace --all-features
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    if: "startsWith(github.ref, 'refs/tags/')"
+    needs: [ check, test ]
+    environment: VALIDATOR_RELEASE
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Fetch Substrait submodule tags
+        working-directory: substrait
+        run: |
+          git fetch --recurse-submodules=no origin +refs/tags/*:refs/tags/*
+          git describe --dirty --tags
+      - name: Local build to populate resource files
+        working-directory: rs
+        run: cargo build
+      - name: Publish substrait-validator-derive
+        working-directory: derive
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+        run: cargo publish --allow-dirty
+      - name: Wait for substrait-validator-derive to appear in the index
+        continue-on-error: true
+        run: python3 ci/crates-io-wait.py substrait-validator-derive `cat ci/version` 1800
+      - name: Publish substrait-validator
+        working-directory: rs
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+        run: cargo publish --allow-dirty
```

Comment:
Why can't these crates be published independently?

Reply:
Because
 - one depends on the other;
 - especially with automated releases it seems like a good idea to always release both to not complicate versioning, even if the derive crate is really unlikely to change;
 - cargo won't let you publish a crate if a dependency is not yet in the registry;
 - cargo publish exits with success before the published crate is synchronized into the crate registry (this can take several minutes depending on queue length).

The latter is an [open issue](https://github.com/rust-lang/cargo/issues/9507). Most people use shell scripts, I strongly prefer Python personally (if only because it's already a dependency, and bash is not). Once cargo resolves this, the blocking Python script can be burned at the stake.